### PR TITLE
React js

### PR DESCRIPTION
**Pull request by urbashi123**
#1320 
@alodahl Please review the PR?

### CODE REVIEW

File: javascript/Javascript.md [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+Guided Practice:15 min
+Check for Understanding:20 min
+
+## Prerequisites
+
+There is not touch prerequisites for learning JavaScript. Here are some of the prerequisite for learning JavaScript:
+
++ Basic programming knowledge. Knowledge of core Java sufficient.
+
++ You should have knowledge of HTML.Check [HTML tutorials here.](https://www.w3schools.com/html/)
```
Comment (Kundan28) [inline javascript/Javascript.md:16]:
```suggestion
+ You should have knowledge of HTML. Check [HTML tutorials here.](https://www.w3schools.com/html/)
```
And we already have `web/html.md` for HTML so I don't think that we should link some external link for the same.

File: javascript/Javascript.md [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+
+### Motivation
+
+Why it is worth learning JavaScript?
+
++ **Implement Client-Side Scripts:**  The web developers can easily write custom client-side scripts in JavaScript, and integrate the scripts seamlessly into HTML. They can further use JavaScript to make the web pages interactive, respond to users instaneously, and create richer user interfaces, without less server interaction and reduces server load.
```
Comment (Kundan28) [inline javascript/Javascript.md:22]:
```suggestion
+ **Implement Client-Side Scripts:** The web developers can easily write custom client-side scripts in JavaScript, and integrate the scripts seamlessly into HTML. They can further use JavaScript to make the web pages interactive, respond to users instantaneously, and create richer user interfaces, without less server interaction and reduces server load.
```

File: javascript/Javascript.md [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
++ **Responsive Web Design:** Most developers nowadays adopt responsive web design to make a website accessible and look good across multiple browsers, devices and browsers.The developers must use JavaScript to optimize web pages for mobile devices.
++ **Google AMP Project:** Google has initiated the Accelerated Mobile Pages (AMP) project to create websites that deliver richer user experience across varying mobile devices. Google has combined several optimization techniques to increase the loading speed of mobile web pages directly.The programmers have to use JavaScript while optimizing websites for mobile devices with AMP.
+[For more reference](https://medium.com/@mindfiresolutions.usa/how-important-is-javascript-for-modern-web-developers-2854309b9f52)
+
+## Lesson
+React is a JavaScript library (not a framework) that creates user interfaces (UIs) in a predictable and efficient way using declarative code. You can use it to help build single page applications and mobile apps, or to build complex apps if you utilise it with other libraries.
+React works in declarative code.
+[For more reference](https://www.w3schools.com/react/)
```
Comment (Kundan28) [inline javascript/Javascript.md:30]: For react too we have `react-js/react.md`

File: javascript/Javascript.md [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+## Lesson
+React is a JavaScript library (not a framework) that creates user interfaces (UIs) in a predictable and efficient way using declarative code. You can use it to help build single page applications and mobile apps, or to build complex apps if you utilise it with other libraries.
+React works in declarative code.
+[For more reference](https://www.w3schools.com/react/) 
+
+1. [Tutorial Using slide](https://www.slideshare.net/asllani94/introduction-to-react-js-63948385)
+
+##### Example 
+
+![1*u94yCCRukpbbePtqah8ASA](https://miro.medium.com/max/1400/1%2Au94yCCRukpbbePtqah8ASA.png)
```
Comment (Kundan28) [inline javascript/Javascript.md:36]: The label could be more descriptive.

**Comment by urbashi123** [comment]
> @urbashi123 Why you added Spread Operator file ? I don't see anything mentioned related to that in the issue.

I have made a seperate branch and made a PR , I don't know how it came. What should I do know?

**Comment by Kundan28** [comment]
Never create a new PR for the same thing instead try to fix the issue in the same PR. You could search for the issue on web and you would definitely get the results,
https://stackoverflow.com/questions/36168839/how-to-remove-commits-from-a-pull-request

**Pull request closed without merging**